Lib/test/test_generators.py

```diff
@@ -306,6 +306,26 @@ def gen():
         self.assertEqual(next(g), "done")
         self.assertEqual(sys.exc_info(), (None, None, None))
 
+    def test_nested_gen_except_loop(self):
+        def gen():
+            for i in range(100):
+                self.assertEqual(sys.exc_info()[0], TypeError)
+                yield "doing"
+
+        def outer():
+            try:
+                raise TypeError
+            except:
+                for x in gen():
+                    yield x
+
+        try:
+            raise ValueError
+        except Exception:
+            for x in outer():
+                self.assertEqual(x, "doing")
+        self.assertEqual(sys.exc_info(), (None, None, None))
```

Comment:
```suggestion
        self.assertEqual(sys.exception(), None)
```

(it's the same thing).